Legacy/pinsec/hardware: updated (REVIEW/REWORD requested) (#186)
SpinalHDL/SpinalDoc-RTD: source/SpinalHDL/Legacy/pinsec/hardware.rst

AXI4
----

> As previously said, Pinsec integrate an AXI4 bus fabric. AXI4 is not the easiest bus to work with but has many advantages like :

Reviewer: ```suggestion
As previously said, Pinsec integrates an AXI4 bus fabric. AXI4 is not the easiest bus to work with but has many advantages like:
```

* A flexible topology
* High bandwidth potential
* Potential out of order request completion
* Easy methods to meets clocks timings
* Standard used by many IP cores
* A handshake methodology that fits with SpinalHDL Stream.

From an Area utilization perspective, AXI4 is for sure not the lightest solution, but some techniques could dramatically reduce that concern :

* Using Read-Only/Write-Only AXI4 variations where that is possible
* Introducing an Axi4-Shared variation where a new ARW channel is introduced to replace AR and AW channels. This solution divide resources usage by two for the address decoding and the address arbitration.

> * Depending the interconnect implementation, if masters doesn't use the R/B channels ready, this path will be removed until each slaves at synthesis, which relax timings.  REVIEWME

Reviewer: ```suggestion
* Timing relaxation is possible depending the interconnect implementation, and if masters never stall the R/B channel (RREADY and BREADY are strapped to 1). Both xREADY signals can be removed by synthesis in this case, relaxing timings.
```

> * As the AXI4 spec suggest, the interconnect can expand the transactions ID by aggregating the corresponding input port ID. This allow the interconnect to have an infinite number of pending request and also to support out of order completion with a negligible area cost (transaction ID expand).

Reviewer: ```suggestion
* As the AXI4 spec suggest, the interconnect can expand the transactions ID by aggregating the corresponding input port ID. This allows the interconnect to have an infinite number of pending request and also to support out of order completion with a negligible area cost (transaction ID expand).
```
